.github/workflows/kind.yaml

```diff
@@ -69,7 +69,9 @@ jobs:
             --set cni.chainingMode=portmap \
             --set loadBalancer.l7.backend=envoy \
             --set tls.secretsBackend=k8s \
-            --set prometheus.enabled=true
+            --set prometheus.enabled=true \
+            --set localRedirectPolicy=true \
+            --set socketLB.enabled=true
```

Comment:
I suppose this kind cluster is common for all connectivity tests? If that's the case, do we want to enable socket-LB for all tests?

Reply:
socket-LB should be disabled in some tests to ensure we get testing coverage for that case too.

Reply:
cc @cilium/cilium-cli-maintainers if you want to chime in about how we balance the coverage across the github workflows

Reply:
Compared the result of the connectivity test between main and my branch. Looks like no difference in tests run except for `local-redirect-policy-with-node-dns`

main branch
```
ℹ️  Cilium version: 1.15.7
🏃[test-namespace-1] Running 18 tests ...
🏃[test-namespace-2] Running 17 tests ...
🏃[test-namespace-3] Running 16 tests ...
🏃[test-namespace-4] Running 17 tests ...
🏃[test-namespace-5] Running 16 tests ...

[=] [test-namespace-1] Test [no-interrupted-connections] [1/18]
[=] [test-namespace-1] Skipping test [no-ipsec-xfrm-errors] [2/18] (requires Feature encryption-pod mode ipsec, got disabled)
[=] [test-namespace-1] Test [no-unexpected-packet-drops] [3/18]
[=] [test-namespace-1] Test [client-ingress] [4/18]
[=] [test-namespace-1] Test [all-ingress-deny-knp] [5/18]
[=] [test-namespace-1] Skipping test [cluster-entity-multi-cluster] [6/18] (skipped by condition)
[=] [test-namespace-1] Test [echo-ingress-knp] [7/18]
[=] [test-namespace-1] Test [client-egress-expression-knp] [8/18]
[=] [test-namespace-1] Test [to-cidr-external-knp] [9/18]
[=] [test-namespace-1] Test [client-ingress-to-echo-named-port-deny] [10/18]
[=] [test-namespace-1] Test [client-egress-to-cidr-deny-default] [11/18]
[=] [test-namespace-1] Test [node-to-node-encryption] [12/18]
[=] [test-namespace-1] Test [north-south-loadbalancing-with-l7-policy] [13/18]
[=] [test-namespace-1] Test [client-egress-l7-named-port] [14/18]
[=] [test-namespace-1] Skipping test [echo-ingress-mutual-auth-spiffe] [15/18] (Feature mutual-auth-spiffe is disabled)
[=] [test-namespace-1] Skipping test [pod-to-ingress-service-allow-ingress-identity] [16/18] (Feature ingress-controller is disabled)
[=] [test-namespace-1] Test [dns-only] [17/18]
[=] [test-namespace-1] Skipping test [pod-to-k8s-on-controlplane-cidr] [18/18] (skipped by condition)

[=] [test-namespace-2] Test [no-policies] [1/17]
[=] [test-namespace-2] Test [client-ingress-knp] [2/17]
[=] [test-namespace-2] Test [all-egress-deny] [3/17]
[=] [test-namespace-2] Test [host-entity-egress] [4/17]
[=] [test-namespace-2] Test [client-ingress-icmp] [5/17]
[=] [test-namespace-2] Test [client-with-service-account-egress-to-echo] [6/17]
[=] [test-namespace-2] Test [from-cidr-host-netns] [7/17]
[=] [test-namespace-2] Test [client-egress-to-echo-expression-deny] [8/17]
[=] [test-namespace-2] Skipping test [clustermesh-endpointslice-sync] [9/17] (skipped by condition)
[=] [test-namespace-2] Skipping test [egress-gateway] [10/17] (Feature enable-ipv4-egress-gateway is disabled)
[=] [test-namespace-2] Test [echo-ingress-l7] [11/17]
[=] [test-namespace-2] Skipping test [echo-ingress-l7-via-hostport-with-encryption] [12/17] (skipped by condition)
[=] [test-namespace-2] Test [client-egress-l7-tls-deny-without-headers] [13/17]
[=] [test-namespace-2] Skipping test [pod-to-ingress-service] [14/17] (Feature ingress-controller is disabled)
[=] [test-namespace-2] Skipping test [outside-to-ingress-service] [15/17] (Feature ingress-controller is disabled)
[=] [test-namespace-2] Test [to-fqdns] [16/17]
[=] [test-namespace-2] Skipping test [local-redirect-policy] [17/17] (Feature enable-local-redirect-policy is disabled)

[=] [test-namespace-3] Test [no-policies-from-outside] [1/16]
[=] [test-namespace-3] Test [allow-all-with-metrics-check] [2/16]
[=] [test-namespace-3] Test [all-egress-deny-knp] [3/16]
[=] [test-namespace-3] Test [host-entity-ingress] [4/16]
[=] [test-namespace-3] Test [client-egress] [5/16]
[=] [test-namespace-3] Test [client-egress-to-echo-service-account] [6/16]
[=] [test-namespace-3] Test [echo-ingress-from-other-client-deny] [7/16]
[=] [test-namespace-3] Test [client-with-service-account-egress-to-echo-deny] [8/16]
[=] [test-namespace-3] Test [health] [9/16]
[=] [test-namespace-3] Skipping test [egress-gateway-excluded-cidrs] [10/16] (Feature enable-ipv4-egress-gateway is disabled)
[=] [test-namespace-3] Test [echo-ingress-l7-named-port] [11/16]
[=] [test-namespace-3] Test [client-egress-l7-tls-headers] [12/16]
[=] [test-namespace-3] Skipping test [pod-to-ingress-service-deny-all] [13/16] (Feature ingress-controller is disabled)
[=] [test-namespace-3] Skipping test [outside-to-ingress-service-deny-world-identity] [14/16] (Feature ingress-controller is disabled)
[=] [test-namespace-3] Skipping test [pod-to-controlplane-host] [15/16] (skipped by condition)
[=] [test-namespace-3] Test [pod-to-pod-no-frag] [16/16]

[=] [test-namespace-4] Test [no-policies-extra] [1/17]
[=] [test-namespace-4] Test [all-ingress-deny] [2/17]
[=] [test-namespace-4] Test [all-entities-deny] [3/17]
[=] [test-namespace-4] Test [echo-ingress] [4/17]
[=] [test-namespace-4] Test [client-egress-knp] [5/17]
[=] [test-namespace-4] Test [to-entities-world] [6/17]
[=] [test-namespace-4] Test [client-ingress-from-other-client-icmp-deny] [7/17]
[=] [test-namespace-4] Test [client-egress-to-echo-service-account-deny] [8/17]
[=] [test-namespace-4] Test [north-south-loadbalancing] [9/17]
[=] [test-namespace-4] Skipping test [egress-gateway-with-l7-policy] [10/17] (skipped by condition)
[=] [test-namespace-4] Test [client-egress-l7-method] [11/17]
[=] [test-namespace-4] Test [client-egress-l7-set-header] [12/17]
[=] [test-namespace-4] Skipping test [pod-to-ingress-service-deny-ingress-identity] [13/17] (Feature ingress-controller is disabled)
[=] [test-namespace-4] Skipping test [outside-to-ingress-service-deny-cidr] [14/17] (Feature ingress-controller is disabled)
[=] [test-namespace-4] Skipping test [pod-to-k8s-on-controlplane] [15/17] (skipped by condition)
[=] [test-namespace-4] Skipping test [bgp-control-plane-v1] [16/17] (Feature enable-bgp-control-plane is disabled)
[=] [test-namespace-4] Skipping test [bgp-control-plane-v2] [17/17] (Feature enable-bgp-control-plane is disabled)

[=] [test-namespace-5] Test [allow-all-except-world] [1/16]
[=] [test-namespace-5] Test [all-ingress-deny-from-outside] [2/16]
[=] [test-namespace-5] Test [cluster-entity] [3/16]
[=] [test-namespace-5] Test [echo-ingress-from-outside] [4/16]
[=] [test-namespace-5] Test [client-egress-expression] [5/16]
[=] [test-namespace-5] Test [to-cidr-external] [6/16]
[=] [test-namespace-5] Test [client-egress-to-echo-deny] [7/16]
[=] [test-namespace-5] Test [client-egress-to-cidr-deny] [8/16]
[=] [test-namespace-5] Test [pod-to-pod-encryption] [9/16]
[=] [test-namespace-5] Skipping test [pod-to-pod-with-l7-policy-encryption] [10/16] (requires Feature encryption-pod mode wireguard, got disabled)
[=] [test-namespace-5] Skipping test [pod-to-node-cidrpolicy] [11/16] (Feature cidr-match-nodes is disabled)
[=] [test-namespace-5] Test [client-egress-l7] [12/16]
[=] [test-namespace-5] Skipping test [echo-ingress-auth-always-fail] [13/16] (Feature mutual-auth-spiffe is disabled)
[=] [test-namespace-5] Skipping test [pod-to-ingress-service-deny-backend-service] [14/16] (Feature ingress-controller is disabled)
[=] [test-namespace-5] Skipping test [outside-to-ingress-service-deny-all-ingress] [15/16] (Feature ingress-controller is disabled)
[=] [test-namespace-5] Skipping test [pod-to-controlplane-host-cidr] [16/16] (skipped by condition)

✅ [test-namespace-1] All 13 tests (85 actions) successful, 5 tests skipped, 0 scenarios skipped.

✅ [test-namespace-2] All 11 tests (136 actions) successful, 6 tests skipped, 0 scenarios skipped.

✅ [test-namespace-3] All 12 tests (86 actions) successful, 4 tests skipped, 0 scenarios skipped.

✅ [test-namespace-4] All 11 tests (99 actions) successful, 6 tests skipped, 0 scenarios skipped.

✅ [test-namespace-5] All 10 tests (77 actions) successful, 6 tests skipped, 0 scenarios skipped.

🏃[test-namespace-1] Running 3 tests ...
[=] [test-namespace-1] Skipping test [host-firewall-ingress] [1/3] (Feature host-firewall is disabled)
[=] [test-namespace-1] Skipping test [host-firewall-egress] [2/3] (Feature host-firewall is disabled)
[=] [test-namespace-1] Test [check-log-errors] [3/3]
..................
✅ [test-namespace-1] All 1 tests (18 actions) successful, 2 tests skipped, 0 scenarios skipped.

```

pr/ysksuzuki/lrp-node-dns branch
```
ℹ️  Cilium version: 1.15.7
🏃[test-namespace-1] Running 18 tests ...
🏃[test-namespace-2] Running 17 tests ...
🏃[test-namespace-3] Running 16 tests ...
🏃[test-namespace-4] Running 16 tests ...
🏃[test-namespace-5] Running 18 tests ...

[=] [test-namespace-1] Test [no-interrupted-connections] [1/18]
[=] [test-namespace-1] Skipping test [no-ipsec-xfrm-errors] [2/18] (requires Feature encryption-pod mode ipsec, got disabled)
[=] [test-namespace-1] Test [no-unexpected-packet-drops] [3/18]
[=] [test-namespace-1] Test [client-ingress] [4/18]
[=] [test-namespace-1] Test [all-ingress-deny-knp] [5/18]
[=] [test-namespace-1] Skipping test [cluster-entity-multi-cluster] [6/18] (skipped by condition)
[=] [test-namespace-1] Test [echo-ingress-knp] [7/18]
[=] [test-namespace-1] Test [client-egress-expression-knp] [8/18]
[=] [test-namespace-1] Test [to-cidr-external-knp] [9/18]
[=] [test-namespace-1] Test [client-ingress-to-echo-named-port-deny] [10/18]
[=] [test-namespace-1] Test [client-egress-to-cidr-deny-default] [11/18]
[=] [test-namespace-1] Test [node-to-node-encryption] [12/18]
[=] [test-namespace-1] Test [north-south-loadbalancing-with-l7-policy] [13/18]
[=] [test-namespace-1] Test [client-egress-l7-named-port] [14/18]
[=] [test-namespace-1] Skipping test [echo-ingress-mutual-auth-spiffe] [15/18] (Feature mutual-auth-spiffe is disabled)
[=] [test-namespace-1] Skipping test [pod-to-ingress-service-allow-ingress-identity] [16/18] (Feature ingress-controller is disabled)
[=] [test-namespace-1] Test [dns-only] [17/18]
[=] [test-namespace-1] Skipping test [pod-to-k8s-on-controlplane-cidr] [18/18] (skipped by condition)

[=] [test-namespace-2] Test [no-policies] [1/17]
[=] [test-namespace-2] Test [client-ingress-knp] [2/17]
[=] [test-namespace-2] Test [all-egress-deny] [3/17]
[=] [test-namespace-2] Test [host-entity-egress] [4/17]
[=] [test-namespace-2] Test [client-ingress-icmp] [5/17]
[=] [test-namespace-2] Test [client-with-service-account-egress-to-echo] [6/17]
[=] [test-namespace-2] Test [from-cidr-host-netns] [7/17]
[=] [test-namespace-2] Test [client-egress-to-echo-expression-deny] [8/17]
[=] [test-namespace-2] Skipping test [clustermesh-endpointslice-sync] [9/17] (skipped by condition)
[=] [test-namespace-2] Skipping test [egress-gateway] [10/17] (Feature enable-ipv4-egress-gateway is disabled)
[=] [test-namespace-2] Test [echo-ingress-l7] [11/17]
[=] [test-namespace-2] Skipping test [echo-ingress-l7-via-hostport-with-encryption] [12/17] (skipped by condition)
[=] [test-namespace-2] Test [client-egress-l7-tls-deny-without-headers] [13/17]
[=] [test-namespace-2] Skipping test [pod-to-ingress-service] [14/17] (Feature ingress-controller is disabled)
[=] [test-namespace-2] Skipping test [outside-to-ingress-service] [15/17] (Feature ingress-controller is disabled)
[=] [test-namespace-2] Test [to-fqdns] [16/17]
[=] [test-namespace-2] Skipping test [local-redirect-policy] [17/17] (skipped by condition)

[=] [test-namespace-3] Test [no-policies-from-outside] [1/16]
[=] [test-namespace-3] Test [allow-all-with-metrics-check] [2/16]
[=] [test-namespace-3] Test [all-egress-deny-knp] [3/16]
[=] [test-namespace-3] Test [host-entity-ingress] [4/16]
[=] [test-namespace-3] Test [client-egress] [5/16]
[=] [test-namespace-3] Test [client-egress-to-echo-service-account] [6/16]
[=] [test-namespace-3] Test [echo-ingress-from-other-client-deny] [7/16]
[=] [test-namespace-3] Test [client-with-service-account-egress-to-echo-deny] [8/16]
[=] [test-namespace-3] Test [health] [9/16]
[=] [test-namespace-3] Skipping test [egress-gateway-excluded-cidrs] [10/16] (Feature enable-ipv4-egress-gateway is disabled)
[=] [test-namespace-3] Test [echo-ingress-l7-named-port] [11/16]
[=] [test-namespace-3] Test [client-egress-l7-tls-headers] [12/16]
[=] [test-namespace-3] Skipping test [pod-to-ingress-service-deny-all] [13/16] (Feature ingress-controller is disabled)
[=] [test-namespace-3] Skipping test [outside-to-ingress-service-deny-world-identity] [14/16] (Feature ingress-controller is disabled)
[=] [test-namespace-3] Skipping test [pod-to-controlplane-host] [15/16] (skipped by condition)
[=] [test-namespace-3] Test [local-redirect-policy-with-node-dns] [16/16]

[=] [test-namespace-4] Test [no-policies-extra] [1/16]
[=] [test-namespace-4] Test [all-ingress-deny] [2/16]
[=] [test-namespace-4] Test [all-entities-deny] [3/16]
[=] [test-namespace-4] Test [echo-ingress] [4/16]
[=] [test-namespace-4] Test [client-egress-knp] [5/16]
[=] [test-namespace-4] Test [to-entities-world] [6/16]
[=] [test-namespace-4] Test [client-ingress-from-other-client-icmp-deny] [7/16]
[=] [test-namespace-4] Test [client-egress-to-echo-service-account-deny] [8/16]
[=] [test-namespace-4] Test [north-south-loadbalancing] [9/16]
[=] [test-namespace-4] Skipping test [egress-gateway-with-l7-policy] [10/16] (skipped by condition)
[=] [test-namespace-4] Test [client-egress-l7-method] [11/16]
[=] [test-namespace-4] Test [client-egress-l7-set-header] [12/16]
[=] [test-namespace-4] Skipping test [pod-to-ingress-service-deny-ingress-identity] [13/16] (Feature ingress-controller is disabled)
[=] [test-namespace-4] Skipping test [outside-to-ingress-service-deny-cidr] [14/16] (Feature ingress-controller is disabled)
[=] [test-namespace-4] Skipping test [pod-to-k8s-on-controlplane] [15/16] (skipped by condition)
[=] [test-namespace-4] Test [pod-to-pod-no-frag] [16/16]

[=] [test-namespace-5] Test [allow-all-except-world] [1/18]
[=] [test-namespace-5] Test [all-ingress-deny-from-outside] [2/18]
[=] [test-namespace-5] Test [cluster-entity] [3/18]
[=] [test-namespace-5] Test [echo-ingress-from-outside] [4/18]
[=] [test-namespace-5] Test [client-egress-expression] [5/18]
[=] [test-namespace-5] Test [to-cidr-external] [6/18]
[=] [test-namespace-5] Test [client-egress-to-echo-deny] [7/18]
[=] [test-namespace-5] Test [client-egress-to-cidr-deny] [8/18]
[=] [test-namespace-5] Test [pod-to-pod-encryption] [9/18]
[=] [test-namespace-5] Skipping test [pod-to-pod-with-l7-policy-encryption] [10/18] (requires Feature encryption-pod mode wireguard, got disabled)
[=] [test-namespace-5] Skipping test [pod-to-node-cidrpolicy] [11/18] (Feature cidr-match-nodes is disabled)
[=] [test-namespace-5] Test [client-egress-l7] [12/18]
[=] [test-namespace-5] Skipping test [echo-ingress-auth-always-fail] [13/18] (Feature mutual-auth-spiffe is disabled)
[=] [test-namespace-5] Skipping test [pod-to-ingress-service-deny-backend-service] [14/18] (Feature ingress-controller is disabled)
[=] [test-namespace-5] Skipping test [outside-to-ingress-service-deny-all-ingress] [15/18] (Feature ingress-controller is disabled)
[=] [test-namespace-5] Skipping test [pod-to-controlplane-host-cidr] [16/18] (skipped by condition)
[=] [test-namespace-5] Skipping test [bgp-control-plane-v1] [17/18] (Feature enable-bgp-control-plane is disabled)
[=] [test-namespace-5] Skipping test [bgp-control-plane-v2] [18/18] (Feature enable-bgp-control-plane is disabled)

✅ [test-namespace-1] All 13 tests (85 actions) successful, 5 tests skipped, 0 scenarios skipped.

✅ [test-namespace-2] All 11 tests (136 actions) successful, 6 tests skipped, 0 scenarios skipped.

✅ [test-namespace-3] All 12 tests (88 actions) successful, 4 tests skipped, 0 scenarios skipped.

✅ [test-namespace-4] All 12 tests (100 actions) successful, 4 tests skipped, 0 scenarios skipped.

✅ [test-namespace-5] All 10 tests (77 actions) successful, 8 tests skipped, 0 scenarios skipped.
🏃[test-namespace-1] Running 3 tests ...
[=] [test-namespace-1] Skipping test [host-firewall-ingress] [1/3] (Feature host-firewall is disabled)
[=] [test-namespace-1] Skipping test [host-firewall-egress] [2/3] (Feature host-firewall is disabled)
[=] [test-namespace-1] Test [check-log-errors] [3/3]
..................
✅ [test-namespace-1] All 1 tests (18 actions) successful, 2 tests skipped, 0 scenarios skipped.

```